sphinx_js/typedoc.py

```diff
@@ -0,0 +1,277 @@
+                for type in node.get('extendedTypes',[]):
+                    doclet['classdesc'] += ' * :js:class:`' + self.make_type_name(type) + '`\n'
+            if node.get('implementedTypes'):
+                doclet['classdesc'] += '\n\n**Implements:**\n'
+                for type in node.get('implementedTypes',[]):
+                    doclet['classdesc'] += ' * :js:class:`' + self.make_type_name(type) + '`\n'
+
+            doclet['params'] = []
+            for param in node.get('typeParameter',[]):
+                doclet['params'].append(self.make_param(param))
+            self.extend_doclet(doclet,
+                extends = [ e['name'] for e in node.get('extendedTypes',[]) ]
+            )
+
+        elif kindString == 'Property':
+            doclet = self.simple_doclet('member', node)
+            if node.get('flags',{}).get('isAbstract'):
+                doclet['description'] = '*abstract*\n\n' + doclet['description']
+            self.extend_doclet(doclet,
+                type = self.make_type(node.get('type'))
+            )
+
+        elif kindString == 'Accessor':
+            doclet = self.simple_doclet('member', node)
+            if node.get('getSignature'):
+                type = self.make_type(node['getSignature']['type'])
+            else:
+                type = self.make_type(node['setSignature']['parameters'][0]['type'])
+            self.extend_doclet(doclet, type=type)
+
+        elif kindString in ['Function', 'Constructor', 'Method']:
+            for sig in node.get('signatures'):
+                sig['sources'] = node['sources']
+                self.convert_node(sig)
+            return
+
+        elif kindString in ['Constructor signature', 'Call signature']:
+            parent = self.get_parent(node)
+            doclet = self.simple_doclet('function', node)
+            if parent.get('flags',{}).get('isAbstract'):
+                doclet['description'] = '*abstract*\n\n' + doclet['description']
+            if parent.get('flags',{}).get('isOptional'):
+                doclet['description'] = '*optional*\n\n' + doclet['description']
+            self.extend_doclet(doclet,
+                params = [],
+                returns = self.make_result(node)
+            )
+            doclet['meta']['code']['paramnames'] = []
+            for param in node.get('parameters',[]):
+                doclet['params'].append(self.make_param(param))
+                doclet['meta']['code']['paramnames'].append(param.get('name'))
+        else:
+            doclet = None
+        if doclet:
+            self.jsdoc.append(doclet)
+        for child in node.get('children', []):
+            self.convert_node(child)
+
+    def make_node_list(self, node, parent=None):
+        if node==None:
+            return
+        if node.get('id')!=None:
+            node['__parentId'] = parent
+            self.nodelist[node['id']] = node
+        for tag in [ 'children', 'signatures', 'parameters']:
```

Comment:
There are some errant and missing spaces here and there. Just run flake8 over this file, and they'll pop up. (Sorry I don't have that set up in the project yet.)